qt/python/mantidqt/utils/qt/test/gui_test_runner.py

```diff
@@ -0,0 +1,197 @@
+                    return
+                # Run test script until the next 'yield'
+                ret = self.script_iter.next()
+                if ret is not None:
+                    if inspect.isgenerator(ret):
+                        self.parent_iter = self.script_iter
+                        self.script_iter = ret
+                    else:
+                        # Start non-blocking pause in seconds
+                        self.pause_timer.start(int(ret * 1000))
+            except StopIteration:
+                if self.parent_iter is not None:
+                    self.script_iter = self.parent_iter
+                    self.parent_iter = None
+                else:
+                    self.script_iter = None
+                    self.script_timer.stop()
+                    if self.close_on_finish:
+                        QMetaObject.invokeMethod(self.widget, 'close', Qt.QueuedConnection)
+            except Exception as e:
+                self.widget.close()
+                traceback.print_exc()
+                self.error = e
+
+
+def open_in_window(widget_or_name, script, attach_debugger=True, pause=0, close_on_finish=False, is_cli=False):
+    """
+    Displays a widget in a window.
+    :param widget_or_name: A widget to display.
+            1. If a string it's a qualified name of a widget, eg mantidqt.mywidget.MyWidget
+            2. If a callable it must take no arguments and return a QWidget
+            3. If a QWidget it's used directly.
+    :param script: A script to run after the widget is open.
+            1. If a string it's a qualified name of a test function that can be run after the
+            widget is created.
+            2. If a callable it's used directly.
+
+        The test function must have the signature:
+
+            def test(widget):
+                ...
+
+        where argument widget is an instance of the tested widget.
+        The test function can yield from time to time after which the widget can update itself.
+        This will make the test non-blocking and changes can be viewed as the script runs.
+        If the test yields an number it is interpreted as the number of seconds to wait
+        until the next step.
+        The test can yield a generator. In this case it will be iterated over until it stops
+        and the iterations of the main script continue.
+    :param attach_debugger: If true pause to let the user to attache a debugger before starting
+        application.
+    :param pause: A number of seconds to wait between the iterations.
+    :param close_on_finish: An option to close the widget after the script finishes.
+    :param is_cli: If true the script is to be run from a command line tool. Exceptions are
+        treated slightly differently in this case.
+    """
+    global app
+    if attach_debugger:
+        raw_input('Please attach the Debugger now if required. Press any key to continue')
+    if app is None:
+        setup_library_paths()
```

Comment:
We've duplicated this code a couple of times. Should we move it to a function in `test/__init__.py` so that it can all reference the existing `_QAPP` object?